metadata/src/main/java/org/apache/kafka/metadata/migration/KRaftMigrationZkWriter.java

```diff
@@ -260,37 +293,37 @@ public void visitScramCredential(String userName, ScramMechanism scramMechanism,
         });
 
         changedNonUserEntities.forEach(entity -> {
-            Map<String, Double> quotaMap = clientQuotasImage.entities().get(entity).quotaMap();
-            operationConsumer.accept("Update client quotas for " + entity, migrationState ->
+            Map<String, Double> quotaMap = getClientQuotaMapForEntity(clientQuotasImage, entity);
+            opConsumer.accept(UPDATE_CLIENT_QUOTA, "Update client quotas for " + entity, migrationState ->
                 migrationClient.configClient().writeClientQuotas(entity.entries(), quotaMap, Collections.emptyMap(), migrationState));
         });
 
         changedUsers.forEach(userName -> {
             ClientQuotaEntity entity = new ClientQuotaEntity(Collections.singletonMap(ClientQuotaEntity.USER, userName));
-            Map<String, Double> quotaMap = clientQuotasImage.entities().get(entity).quotaMap();
+            Map<String, Double> quotaMap = getClientQuotaMapForEntity(clientQuotasImage, entity);
```

Comment:
There was an NPE here prior to this patch. If the SCRAM credentials changed, but ClientQuotas did not, the `get(entity)` call would return a null.